src/test/kotlin/com/exactpro/th2/http/client/ApplicationIntegrationTest.kt

```diff
@@ -0,0 +1,353 @@
+import com.exactpro.th2.test.spec.RabbitMqSpec
+import com.exactpro.th2.test.spec.RabbitMqSpec.Companion.EVENTS_PIN_NAME
+import com.exactpro.th2.test.spec.filter
+import com.exactpro.th2.test.spec.message
+import com.exactpro.th2.test.spec.pin
+import com.exactpro.th2.test.spec.pins
+import com.exactpro.th2.test.spec.publishers
+import com.exactpro.th2.test.spec.subscribers
+import org.junit.jupiter.api.Test
+import strikt.api.Assertion
+import strikt.api.expectThat
+import strikt.assertions.all
+import strikt.assertions.hasSize
+import strikt.assertions.isEmpty
+import strikt.assertions.isEqualTo
+import strikt.assertions.isFalse
+import strikt.assertions.matches
+import strikt.assertions.withElementAt
+import java.time.Duration.ofSeconds
+import java.time.Instant
+import kotlin.test.assertNotNull
+
+@IntegrationTest
+@Th2IntegrationTest
+class ApplicationIntegrationTest {
+    @JvmField
+    @Suppress("unused")
+    internal val customConfig = CustomConfigSpec.fromString(
+        """
+        {
+            "host": "127.0.0.1",
+            "port": 8080,
+            "sessionAlias": "some_api",
+            "validateCertificates": false,
+            "sessionAlias": "test-session-alias",
+            "useTransport": true
+        }
+        """.trimIndent()
+    )
+
+    @JvmField
+    @Suppress("unused")
+    internal val mq = RabbitMqSpec.create()
+        .pins {
+            subscribers {
+                pin("sub") {
+                    attributes(INPUT_QUEUE_TRANSPORT_ATTRIBUTE, TRANSPORT_GROUP_ATTRIBUTE)
+                    filter {
+                        message {
+                            field("test") shouldBeEqualTo "a"
+                        }
+                    }
+                }
+            }
+
+            publishers {
+                pin("pub") {
+                    attributes(TRANSPORT_GROUP_ATTRIBUTE)
+                }
+            }
+        }
+
+    @Test
+    fun `failed connection when process message without parent event id test`(
+        @Th2AppFactory appFactory: CommonFactory,
+        @Th2TestFactory testFactory: CommonFactory,
+        resources: CleanupExtension.Registry,
+    ) {
+        val eventListener = CollectorMessageListener.createWithCapacity<EventBatch>(1)
+        testFactory.eventBatchRouter.subscribe(eventListener, EVENTS_PIN_NAME)
+
+        val application = Application(appFactory) { resource, destructor ->
+            resources.add(resource, destructor)
+        }
+
+        val rootEventId: EventID = eventListener.assertRootEvent().id
+
+        application.start()
+
+        testFactory.sendMessages(RawMessage.builder().apply {
+            idBuilder().apply {
+                setSessionAlias("test-session-alias")
+                setTimestamp(Instant.now())
+                setDirection(Direction.OUTGOING)
+                setSequence(1)
+            }
+        }.build())
+
+        assertNotNull(eventListener.poll(ofSeconds(2))).also {
```

Comment:
You should use only one assertion library. Mixing standard junit assertion with Strikt assertion does not look good in my opinion. Especially, considering the fact the Strikt library has everything you need
```suggestion
        expectThat(eventListener.poll(ofSeconds(2))).isNotNull().get { /*...*/ }
```

Reply:
expectThat(...) expect lambda as the second argument

Reply:
It has an overload that has only one parameter. Please, check the API one more time

Reply:
I found, thanks